src/EnergyPlus/api/datatransfer.cc

```diff
@@ -141,6 +152,12 @@ void resetErrorFlag(EnergyPlusState state) {
     thisState->dataPluginManager->apiErrorFlag = false;
 }
 
+int getNumNodesInCondFDSurfaceLayer(EnergyPlusState state, const char* surfName, const char* matName) {
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto UCsurfName = EnergyPlus::UtilityRoutines::MakeUPPERCase(surfName);
+    auto UCmatName = EnergyPlus::UtilityRoutines::MakeUPPERCase(matName);
+    return EnergyPlus::HeatBalFiniteDiffManager::numNodesInMaterialLayer(*thisState, UCsurfName, UCmatName);
+}
```

Comment:
Nice. Yes, a new API method to get the number of nodes in a specific material layer.